crates/lib3h/src/engine/real_engine.rs

```diff
@@ -191,7 +189,7 @@ impl<T: Transport, D: Dht> NetworkEngine for RealEngine<T, D> {
 }
 
 /// Drop
-impl<T: Transport, D: Dht> Drop for RealEngine<T, D> {
+impl<'engine, D: Dht> Drop for RealEngine<'engine, D> {
     fn drop(&mut self) {
         let res = self.shutdown();
         if let Err(e) = res {
```

Comment:
What do you think about this one liner instead ? : 
```rust
        self.shutdown().unwrap_or_else(|e| warn!("Graceful shutdown failed: {}", e));
```

Reply:
didn't actually touch this code here, but updated : )